### PR TITLE
Add OXML parse options (symbolize_keys, strip_whitespace, normalize_whitespace, force_utf8, preserve_white_space)

### PR DESCRIPTION
**Pull request by evgenlytkin**
This pull request introduces version 0.6.0 of the `oxml` gem, adding several new parsing options for greater control over key symbolization and whitespace handling, as well as optimizations for large XML inputs. It also improves utility methods and expands test coverage to ensure correctness of the new features.

**New Parsing Options and Features:**

* Added new options: `:symbolize_keys`, `:strip_whitespace`, `:normalize_whitespace`, `:force_utf8`, and `:preserve_white_space` to the parser, allowing users to control key symbolization, whitespace handling, and UTF-8 enforcement. These are now documented in the `README.md` and `CHANGELOG.md`. [[1]](diffhunk://#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR3-R14) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R15-R19) [[3]](diffhunk://#diff-3f7ed4e7082b592a15afb09f82d05514985aae0e907e37425cb6fe1944a2fc48R22-R25) [[4]](diffhunk://#diff-3f7ed4e7082b592a15afb09f82d05514985aae0e907e37425cb6fe1944a2fc48L47-R56) [[5]](diffhunk://#diff-3f7ed4e7082b592a15afb09f82d05514985aae0e907e37425cb6fe1944a2fc48R75-R85) [[6]](diffhunk://#diff-3f7ed4e7082b592a15afb09f82d05514985aae0e907e37425cb6fe1944a2fc48R98-R103)

* For large XML strings (over 0.5MB), parsing now uses `StringIO` to reduce memory allocation, improving performance for big documents. [[1]](diffhunk://#diff-53cece32b2e546fe3a82a3127245a856b87f92fbcd929327c548fcd19b55a348R3) [[2]](diffhunk://#diff-53cece32b2e546fe3a82a3127245a856b87f92fbcd929327c548fcd19b55a348R12-R39)

**Utility Improvements:**

* Improved the `Utils.snakecase` and `Utils.camelize` methods to handle nil and non-string input gracefully, and added corresponding specs to ensure robust behavior. [[1]](diffhunk://#diff-ad7bde4915ba16b23eb240b426a4cb0952f092b341ffcade8a12e51e141cea1cR8-R9) [[2]](diffhunk://#diff-ad7bde4915ba16b23eb240b426a4cb0952f092b341ffcade8a12e51e141cea1cL16-R23) [[3]](diffhunk://#diff-41b5a80029a16d7ba43537f1dd99ac210ef3533d9a08febb70ef144771cb800aR1-R53)

**Testing and Documentation:**

* Added and updated RSpec tests to cover all new options and edge cases, including tests for large XML parsing, key symbolization, whitespace options, and utility methods. [[1]](diffhunk://#diff-5da335a86584c39332d63529a38ddc6833ea057697baf7712bee8fb55af64a27L140-R140) [[2]](diffhunk://#diff-5da335a86584c39332d63529a38ddc6833ea057697baf7712bee8fb55af64a27R305-R318) [[3]](diffhunk://#diff-5da335a86584c39332d63529a38ddc6833ea057697baf7712bee8fb55af64a27R340-R401) [[4]](diffhunk://#diff-41b5a80029a16d7ba43537f1dd99ac210ef3533d9a08febb70ef144771cb800aR1-R53)

**Version Bump:**

* Updated the gem version to `0.6.0`.

### CODE REVIEW

File: lib/oxml.rb [hunk 1]
```diff
@@ -8,16 +9,27 @@
...
   def build(hash)
     Builder.new(hash).to_s
   end
+
+  # Use StringIO for large strings to reduce memory allocation
+  def optimize_xml_input(xml)
+    return xml unless xml.is_a?(String) && xml.bytesize > IO_OPTIMIZATION_THRESHOLD
+    
+    StringIO.new(xml)
+  end
```
Comment (xkwd) [inline lib/oxml.rb:39]: Would you be able to add a test for when this threshold is met to make sure we have spec better spec coverage?

File: Rakefile [hunk 2]
```diff
@@ -0,0 +1 @@
+require "bundler/gem_tasks"
```
Comment (xkwd) [inline Rakefile]: Is it a leftover? I don't see any tasks or anything related 🤔

File: oxml.gemspec [hunk 3]
```diff
@@ -13,6 +13,7 @@ Gem::Specification.new do |spec|
   spec.files = Dir['lib/**/*']
   spec.required_ruby_version = '>= 2.6'
   spec.add_dependency 'ox', '~> 2.14'
+  spec.add_development_dependency 'rake', '~> 13.2'
```
Comment (xkwd) [inline oxml.gemspec]: Also a leftover?

File: lib/oxml/parser.rb [hunk 4]
```diff
@@ -19,6 +19,9 @@ def initialize(options = {})
       @delete_namespace_attributes = options.fetch(:delete_namespace_attributes, false)
       @advanced_typecasting = options.fetch(:advanced_typecasting, false)
       @skip_soap_elements = options.fetch(:skip_soap_elements, false)
+      @symbolize_keys = options.fetch(:symbolize_keys, true)
+      @strip_whitespace = options.fetch(:strip_whitespace, false)
+      @normalize_whitespace = options.fetch(:normalize_whitespace, false)
```
Comment (xkwd) [inline lib/oxml/parser.rb:24]: Would you be able to add spec coverage for these three new options? We already have it for every option above

**Review by xkwd — COMMENTED** [review]
This is a nice addition! 👍🏻  I would appreciate if you could briefly mention the changes in the changelog and, if possible, in the readme, at least the fact that new options are being added

File: lib/oxml/parser.rb [hunk 5]
```diff
@@ -19,6 +19,9 @@ def initialize(options = {})
       @delete_namespace_attributes = options.fetch(:delete_namespace_attributes, false)
       @advanced_typecasting = options.fetch(:advanced_typecasting, false)
       @skip_soap_elements = options.fetch(:skip_soap_elements, false)
+      @symbolize_keys = options.fetch(:symbolize_keys, true)
```
Comment (onli) [inline lib/oxml/parser.rb:22]: @xkwd This option would be very useful in the main version. With symbolized keys, the hash generated can not be easily treated by jsonpath tools like https://github.com/joshbuddy/jsonpath and https://rubygems.org/gems/janeway-jsonpath. The first has an option to work with symbols as keys, but it's off by default, and Janeway does have no such option (at least according to its docs)
